### DISCUSSION TITLE
API task output without parsing strings?

### DISCUSSION THREAD

**Discussion by andhya** [semaphoreui/semaphore#1955] [Q&A]
What's the best way of using the Semaphore API to get the output of a task from each remote host?

I'm building an internal tool that pulls a bunch of tools into one webUI for us to cross-reference them. I'm trying to display the output of the latest completed task from a particular template in a table of the remote hosts, along with the information collected from each remote host by other tools.

Getting the last completed task of that template is simple enough using GET project_id/tasks but it's looking like I'd have to do a lot of string parsing of the response from [GET {task_id}/output](https://www.semui.co/api-docs/#/project/get_project__project_id__tasks__task_id__output) to get what I need..

Is there a better way?

**Comment by pakalov** [comment]
Hi, 

maybe you already solved this, but I noticed that it's available the [GET {task_id}/raw_output](https://semaphoreui.com/api-docs/#/project/get_project__project_id__tasks__task_id__raw_output) resource that might be helpful.
The only difference is that you won't have the timestamps of each raw as the "output" resource.

Hope this helps.

**Reply by andhya** [reply]
Thanks @pakalov I guess that might help skip a bit of the string parsing - but still to select the output by remote host, it's a case of parsing strings and best guesses.. I think the answer might be to start a separate task for each remote host, and so the task id can be matched to the remote host?

**Reply by pakalov** [reply]
Hi @andhya, sorry for the late response.

Imho, I think that you're solution would make more harm than good, because you would have to manage the triggering of the same task for multiple hosts, fetch every output, parse them and eventually perform some kind of aggregation. 

Have you tried the [ansible-output-parser](https://github.com/RockProfile/Ansible-Output-Parser) python package? It seems to be built for this, maybe the use of the [GET {task_id}/raw_output](https://semaphoreui.com/api-docs/#/project/get_project__project_id__tasks__task_id__raw_output) endpoint, although I've never tried it.

The last resort would be to build you're own parser...which I don't think It's actually that bad, because the output of playbook execution is somewhat structured, so you can leverage that to ease the parsing (never tested this too).
